### PR TITLE
CSHARP-3972: BulkDeleteOperation ignores isOrdered parameter

### PR DESCRIPTION
**Pull request by BorisDog**
[EG](https://spruce.mongodb.com/version/61980964d6d80a7a2743552b)
Jira ticket: [CSHARP-3972](https://jira.mongodb.org/browse/CSHARP-3972)

### CODE REVIEW

File: src/MongoDB.Driver.Core/Core/Operations/BulkMixedWriteOperation.cs [hunk 1]
```diff
@@ -249,6 +249,7 @@ private IExecutableInRetryableWriteContext<BulkWriteOperationResult> CreateBulkD
             var requests = batch.Requests.Cast<DeleteRequest>();
             return new BulkDeleteOperation(_collectionNamespace, requests, _messageEncoderSettings)
             {
+                IsOrdered = _isOrdered,
```
Comment (DmitryLukyanov) [inline src/MongoDB.Driver.Core/Core/Operations/BulkMixedWriteOperation.cs:252]: let's add a test for this?

Comment (BorisDog) [reply src/MongoDB.Driver.Core/Core/Operations/BulkMixedWriteOperation.cs:252]: Any ideas how to add simple test for that?

Comment (JamesKovacs) [reply src/MongoDB.Driver.Core/Core/Operations/BulkMixedWriteOperation.cs:252]: Boris and I discussed over Slack. We couldn't come up with a simple test to ensure that this flag is set. If we can come up with a creative way to test, we can open a new ticket.

**Review by JamesKovacs — APPROVED** [review]
LGTM

File: src/MongoDB.Driver.Core/Core/Operations/BulkMixedWriteOperation.cs (hunk 1 above)
Comment (DmitryLukyanov) [reply src/MongoDB.Driver.Core/Core/Operations/BulkMixedWriteOperation.cs:252]: at least we have `BulkMixedWriteOperationTests` :)

Comment (DmitryLukyanov) [reply src/MongoDB.Driver.Core/Core/Operations/BulkMixedWriteOperation.cs:252]: maybe test like `Execute_with_mixed_requests_and_ordered_is_false` can be taken as example

Comment (BorisDog) [reply src/MongoDB.Driver.Core/Core/Operations/BulkMixedWriteOperation.cs:252]: Yup, tried that. Could not come up with simple way to fail a delete request.

**Review by DmitryLukyanov — APPROVED** [review]
LGTM

**Comment by BorisDog** [comment]
[Merged](https://github.com/mongodb/mongo-csharp-driver/commit/fff5e6ae90950264dd82bf0146c584c7a18a63ff)

**Pull request closed without merging**